### PR TITLE
Improve Switzerland

### PR DESCRIPTION
**Pull request by herpaderpaldent**
# Improve Switzerland

This pull requests aims to resolve https://github.com/spatie/holidays/issues/193 and adds various imrovements whilst refactoring the code to make it more maintainable.

Whilst the original implementation relied on wikipedia. This pull request implements the [holidays communicated by the federal bureau of justice FOJ](https://www.bj.admin.ch/dam/bj/de/data/publiservice/service/zivilprozessrecht/kant-feiertage.pdf.download.pdf/kant-feiertage.pdf).

- maintainability: refactor regions/cantons
- refactor: instead of maintaining a large array, the regions/cantons are split in individual functions making them easier to maintain and update. This allows for further inheritance for later implementation of districts and cities.
- support more advanced regional holiday calculations
- add labour day
- add rumantsch as language
- maintainability: use English as default language (not German)
- add more tests

### CODE REVIEW

**Comment by Nielsvanpach** [comment]
> maintainability: use English as default language (not German)

I do prefer it to return the most general language by default.

```php
$holidays = Holidays::for('ch')->get(); // German by default instead of English
```

File: tests/Countries/SwitzerlandTest.php [hunk 1]
```diff
@@ -7,6 +7,42 @@
 use Spatie\Holidays\Exceptions\InvalidRegion;
 use Spatie\Holidays\Holidays;
 
+dataset('cantons', [
```
Comment (Nielsvanpach) [inline tests/Countries/SwitzerlandTest.php:10]: I don't think all cantons need a specific snapshot test. You can only verify the edge cases, which is more explicit.

File: tests/Countries/SwitzerlandTest.php [hunk 2]
```diff
@@ -31,28 +67,127 @@
...
+    });
+
+    it('can translate swiss holidays into', function (string $language) {
+        $holidays = Holidays::for(country: 'ch', locale: $language)->get();
+
+        expect($holidays)
+            ->toBeArray()
+            ->not()->toBeEmpty();
+
+        expect(formatDates($holidays))->toMatchSnapshot();
```
Comment (Nielsvanpach) [inline tests/Countries/SwitzerlandTest.php:112]: No need for a big snapshot tests here. You can just verify if a certain holiday name is present.

File: tests/Countries/SwitzerlandTest.php [hunk 3]
```diff
@@ -31,28 +67,127 @@
...
+    })->with('languages');
+
+    it('does use native language for epiphany', function (string $canton, string $name, string $locale) {
+        CarbonImmutable::setTestNowAndTimezone('2024-01-01');
```
Comment (Nielsvanpach) [inline tests/Countries/SwitzerlandTest.php:116]:
```suggestion
        CarbonImmutable::setTestNow('2024-01-01');
```

Comment (Nielsvanpach) [reply tests/Countries/SwitzerlandTest.php:116]: Can be found multiple times in the tests

**Review by Nielsvanpach — CHANGES_REQUESTED** [review]

**Comment by Nielsvanpach** [comment]
> refactor: instead of maintaining a large array, the regions/cantons are split in individual functions making them easier to maintain and update.

I don't really agree this solution is more readable and easier to maintain

**Comment by Nielsvanpach** [comment]
Thanks for your efforts and feel free to reopen if you're still planning to update this PR!
